x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -544,7 +544,7 @@ export class Authenticator {
     // If there is no session to predict which provider to use first, let's use the order
     // providers are configured in. Otherwise return provider that owns session first, and only then the rest
     // of providers.
-    if (!sessionValue) {
+    if (!sessionValue?.provider?.name) {
```

Comment:
This was actually throwing the runtime error described in the linked issue. Once I fixed this, I ran into the other problem above.

Reply:
Since you fixed the root cause in `isLegacyProviderSession` do we still need to do anything here? It seems `sessionValue` is always a return value of `getSessionValue` that relies on fixed `isLegacyProviderSession`.

Reply:
Yep you're absolutely right. I removed this change.